spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -126,16 +127,30 @@ public DefaultKafkaProducerFactory(Map<String, Object> configs) {
 
 	/**
 	 * Construct a factory with the provided configuration and {@link Serializer}s.
+	 * Also configures a {@link #transactionIdPrefix} as a value from the
+	 * {@link ProducerConfig#TRANSACTIONAL_ID_CONFIG} if provided.
+	 * This config is going to be overridden with a suffix for target {@link Producer} instance.
 	 * @param configs the configuration.
 	 * @param keySerializer the key {@link Serializer}.
 	 * @param valueSerializer the value {@link Serializer}.
 	 */
 	public DefaultKafkaProducerFactory(Map<String, Object> configs,
 			@Nullable Serializer<K> keySerializer,
 			@Nullable Serializer<V> valueSerializer) {
+
 		this.configs = new HashMap<>(configs);
 		this.keySerializer = keySerializer;
 		this.valueSerializer = valueSerializer;
+
+		String txId = (String) this.configs.get(ProducerConfig.TRANSACTIONAL_ID_CONFIG);
+		if (StringUtils.hasText(txId)) {
+			setTransactionIdPrefix(txId);
```

Comment:
Needs to be made final when calling from a CTOR. Perhaps add a protected getter if a subclass wants to sniff the value.

Reply:
It is there. See the change on line 186.

I'm OK to add getter.

Reply:
Doh, I missed it because of all the other noise 😦 sorry.